source/s3_auto_ranged_put.c

```diff
@@ -1058,6 +1161,23 @@ static int s_s3_auto_ranged_put_pause(struct aws_s3_meta_request *meta_request,
     aws_s3_meta_request_lock_synced_data(meta_request);
     struct aws_s3_auto_ranged_put *auto_ranged_put = meta_request->impl;
 
+    /**
+     * generate pause token for put.
+     * current format is json with the following structure
+     * {
+     *      type: String
+     *      multipart_upload_id: String
+     *      partition_size: number
+     *      total_num_parts: number
+     * }
+     */
```

Comment:
Why not add this to public header file?

Reply:
my reasoning for not adding it to public headers would be to avoid giving callers any impression that its ok to parse tokens

Reply:
hmm, iirc, one of the reason about we adding the complexity of listing parts is:
- Consistent with v1 implementations and mental model that customers might already have.

So, if we don't expose the format, the Java SDK cannot adapt it, right?

Reply:
what do you mean by adopting here? To java sdk its a transparent token, they should not be parsing. on java side, they just wrap the token in additional data needed to serialize the request and it operates the same way as v1.